bolt/spark/array.py

```diff
@@ -500,6 +500,41 @@ def key_func(key):
 
         return rdd, shape, split
 
+    def _getmixed(self, index):
+        """
+        Mixed indexing (combines basic and advanced indexes)
+
+        Assumes that only a single advanced index is used, due to the complicated
+        behavior needed to be compatible with NumPy otherwise.
+        """
+        # find the single advanced index
+        index = list(index)
+        loc = where([isinstance(i, (tuple, list, ndarray)) for i in index])[0][0]
+        idx = list(index[loc])
+
+        if isinstance(idx[0], (tuple, list, ndarray)):
+            raise ValueError("When mixing basic and advanced indexing, advanced index must be one-dimensional")
+
+        # handle the single advanced index
+        if loc < self.split:    # advanced index is on a key -- filter and update key
+            def newkey(key):
```

Comment:
i prefer to not use same line comments, and rather reword so the comment goes above the `if...else...` block